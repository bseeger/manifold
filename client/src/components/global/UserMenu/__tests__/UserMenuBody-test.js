import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import UserMenuBody from '../UserMenuBody';

describe("GlobalUserMenuBody Component", () => {

  const props = {
    hideUserMenu: () => {},
    startLogout: () => {},
    showLoginOverlay: () => {},
    visible: false
  };

  it('renders correctly', () => {
    const component = renderer.create(
      <UserMenuBody
        {...props}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has the user-menu class", () => {
    expect(shallow(
      <UserMenuBody
        {...props}
      />
    ).is('.user-menu')).toBe(true)
  });

  it("has the menu-visible class when props.visible is true", () => {
    expect(shallow(
      <UserMenuBody
        {...props}
        visible={true}
      />
    ).is('.menu-visible')).toBe(true)
  });

  it("has the menu-hidden class when props.visible is false", () => {
    expect(shallow(
      <UserMenuBody
        {...props}
        visible={false}
      />
    ).is('.menu-hidden')).toBe(true)
  });

});

