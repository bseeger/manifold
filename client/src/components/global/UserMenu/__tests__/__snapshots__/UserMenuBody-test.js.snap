exports[`GlobalUserMenuBody Component renders correctly 1`] = `
<nav
  className="user-menu menu-hidden">
  <i
    className="tail" />
  <ul>
    <li>
      <a
        href="#"
        onClick={[Function]}>
        <i
          className="manicon manicon-person-pencil" />
        Edit Profile
      </a>
    </li>
    <li>
      <button
        onClick={[Function]}>
        <i
          className="manicon manicon-circle-arrow-out-right" />
        Logout
      </button>
    </li>
  </ul>
</nav>
`;
